exonum-java-binding/core/rust/exonum-java/src/node.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use java_bindings::{Command, Config, EjbCommand, EjbCommandResult};
+
+use java_bindings::exonum::blockchain::{Blockchain, BlockchainBuilder, InstanceCollection};
+use java_bindings::exonum::exonum_merkledb::{Database, RocksDB};
+use java_bindings::exonum::node::{ApiSender, Node, NodeChannel};
+use java_bindings::exonum::runtime::rust::ServiceFactory;
+use std::sync::Arc;
+
+pub fn run_node(command: Command) -> Result<(), failure::Error> {
+    if let EjbCommandResult::EjbRun(config) = command.execute()? {
+        let node = create_node(config)?;
+        node.run()
+    } else {
+        Ok(())
+    }
+}
+
+fn create_node(config: Config) -> Result<Node, failure::Error> {
+    let channel = NodeChannel::new(&config.run_config.node_config.mempool.events_pool_capacity);
+    let blockchain = create_blockchain(&config, &channel)?;
+
+    let node_config = config.run_config.node_config;
+    let node_config_path = config
+        .run_config
+        .node_config_path
+        .to_str()
+        .expect("Cannot convert node_config_path to String")
+        .to_owned();
+
+    Ok(Node::with_blockchain(
+        blockchain,
+        channel,
+        node_config,
```

Comment:
That's funny that it accepts the whole `node_config` yet requires node_config_path
        to be passed separately.

Reply:
Yeah, kind of funny. The latter one is needed for hot-changing node config when adding new peers or smthing like that. I think we need to improve Node construction, @popzxc mentioned the idea of builder pattern for it.